hackathon_site/dashboard/frontend/src/components/teamDetail/TeamPendingOrderTable/TeamPendingOrderTable.test.tsx

```diff
@@ -0,0 +1,71 @@
+import React from "react";
+import { getByTestId, render, screen } from "testing/utils";
+import { mockHardware, mockPendingOrdersInTable } from "testing/mockData";
+import TeamPendingOrderTable from "components/teamDetail/TeamPendingOrderTable/TeamPendingOrderTable";
+import { within, waitFor } from "testing/utils";
+import { fireEvent } from "@testing-library/react";
+
+describe("team pending order table", () => {
+    test("renders team pending order table", () => {
+        const { getByTestId } = render(<TeamPendingOrderTable />);
+        expect(screen.getByText("Requested Items")).toBeInTheDocument();
+
+        //loop through all pending orders
+        for (let i = 0; i < mockPendingOrdersInTable.length; i++) {
+            const currentOrder = mockPendingOrdersInTable[i];
+            //loop through all the hardware in each order
+            for (let j = 0; j < currentOrder.hardwareInTableRow.length; j++) {
+                // renders all hardware names
+                const currentRow = currentOrder.hardwareInTableRow[j];
+                expect(
+                    within(
+                        getByTestId(`table-${currentOrder.id}-${currentRow.id}`)
+                    ).getByText(`${mockHardware[currentRow.id - 1].name}`)
+                ).toBeInTheDocument();
+
+                // renders all hardware model numbers
+                expect(
+                    within(
+                        getByTestId(`table-${currentOrder.id}-${currentRow.id}`)
+                    ).getByText(`${mockHardware[currentRow.id - 1].model_number}`)
+                ).toBeInTheDocument();
+
+                // renders all hardware manufacturers
+                expect(
+                    within(
+                        getByTestId(`table-${currentOrder.id}-${currentRow.id}`)
+                    ).getByText(`${mockHardware[currentRow.id - 1].manufacturer}`)
+                ).toBeInTheDocument();
+            }
+            const submitButton = getByTestId(`complete-button-${currentOrder.id}`);
+            fireEvent.click(submitButton);
+        }
+    });
+
+    test("Complete Order button", async () => {
+        const { getByTestId } = render(<TeamPendingOrderTable />);
+
+        for (let i = 0; i < mockPendingOrdersInTable.length; i++) {
+            const currentOrder = mockPendingOrdersInTable[i];
+            const submitButton = getByTestId(`complete-button-${currentOrder.id}`);
+            fireEvent.click(submitButton);
+            await waitFor(() => {
+                for (let j = 0; j < currentOrder.hardwareInTableRow.length; j++) {
+                    const currentRow = currentOrder.hardwareInTableRow[j];
+                    expect(
+                        screen.getByText(
```

Comment:
ah so this doesn't work because you display an alert and the alert doesn't show up on the DOM.
There's no good way to test what shows up on the screen for this function. I originally was thinking you would pass in a function into the table and you can test it that way but we'll have to change that anyways when we link it to the API.

So yeah just remove this test, we'll implement the tests later ... my bad :(

Reply:
np!